src/functionaltests/java/com/ericsson/ei/utils/JenkinsManager.java

```diff
@@ -0,0 +1,226 @@
+package com.ericsson.ei.utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.apache.http.client.ClientProtocolException;
+import org.json.JSONObject;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+
+import com.ericsson.ei.utils.HttpRequest.HttpMethod;
+
+public class JenkinsManager {
+
+    private static final String JENKINS_JOB_TEMPLATE_FILE_PATH = String.join(File.separator, "src", "functionaltests",
+            "resources", "jenkinsJobTemplate.xml");
+    private static final String DEFAULT_TOKEN = "test";
+    private static final String DEFAULT_SCRIPT = "echo &quot;Test&quot;";
+    private static final String DEFAULT_JOB_NAME = "test_job";
+
+    private String host;
+    private int port;
+    private String encoding;
+    private String crumb;
+
+    /**
+     * Constructor, takes jenkins host, port, username and password.
+     * @param host
+     * @param port
+     * @param username
+     * @param password
+     * @throws UnsupportedEncodingException
+     * @throws URISyntaxException
+     */
+    public JenkinsManager(String host, int port, String username, String password)
+            throws UnsupportedEncodingException, URISyntaxException {
+        String authString = String.join(":", username, password);
+
+        this.host = host;
```

Comment:
We haven't discussed which way we prefer to inject all environmental things yet. One way is to add them in the test properties but it can of course be done in other ways such as flags etc. As long it is injectable into this class like it is now, it will be perfect i think :)

Reply:
It is set by the test that initiates this class.

Reply:
Well, its been decided to have a input configuration which probably will be spring properties based.